packages/variable-editor/src/components/variables/data/variable-utils.ts

```diff
@@ -125,3 +133,74 @@ const enrichVariableWithFileMetadata = (variable: Variable, metadata: string) =>
   }
   return variable;
 };
+
+export const toContent = (variables: Array<Variable>) => {
+  return stringify(new Pair(new Scalar('Variables'), parseVariables(variables)));
+};
+
+const parseVariables = (variables: Array<Variable>) => {
+  const map = new YAMLMap();
+  variables.forEach(variable => map.add(parseVariable(variable)));
+  return map;
+};
+
+const parseVariable = (variable: Variable) => {
+  const variableKey = new Scalar(variable.name);
+  variableKey.commentBefore = parseKeyDescription(variable);
+  let variableValue;
+  if (variable.children.length === 0) {
+    variableValue = new Scalar(parseVariableValue(variable.value));
+  } else {
+    variableValue = parseVariables(variable.children);
+  }
+  return new Pair(variableKey, variableValue);
+};
+
+const parseKeyDescription = (variable: Variable) => {
+  const metadata = variable.metadata;
+  const description = variable.description;
+  if (metadata.type === '') {
+    return description && addSingleLeadingWhitespaceToEachLine(description);
```

Comment:
This seems to be wrong?

Reply:
This is a shortcut for:
```
    if (description) {
      return addSingleLeadingWhitespaceToEachLine(description);
    }
    return;  
```
or even more verbose:
```
    if (description !== '') {
      return addSingleLeadingWhitespaceToEachLine(description);
    }
    return undefined;
```

Gonna change it to make it more readable.

Reply:
Ah I though it is a string concatenation. Sorry my fault that I don't know type script. All good.